### ISSUE TITLE
A three-word username is set for a recovered user after completing the Onboarding flow

### ISSUE THREAD

**Issue by mariia-skrypnyk** [status-im/status-react#21827]
# Bug Report
https://github.com/status-im/status-mobile/pull/21715

## Problem 
[comment]: # (A problem description)
User gets old variant user name that we do not use in current app version anymore

#### Actual behavior

![Image](https://github.com/user-attachments/assets/5a88bcd7-e5ee-4ba8-90e6-b735198ec38d)

#### Expected behavior
User should get standard key name and his backed up from Waku name after relogin

### Reproduction

1. Install app
2. Go to Login -> Recovery phrase
3. End onboarding flow

### Additional Information

- Status version: nightly 16/12
- Operating System: Android, iOS

[comment]: # (Please, add logs/notes if necessary)

**Comment by ulisesmac** [comment]
A side note:

If we logout, the name set is sucessfully recovered. Status-go is returning this three-words name the first time we recover the profile

**Comment by ilmotta** [comment]
@ulisesmac, the best solution would be if we completely remove 3-word name implementation in the client and status-go and if status-go had the responsibility of setting the default profile name during profile creation. That way, there's not even a chance that the wrong 3-word name would be used. This way both mobile & desktop clients will be consistent with each other, at least more easily. The only detail is that if we do change status-go we need to align with Desktop team to see if we won't break them while they are working on their own onboarding simplifications.

**Comment by ulisesmac** [comment]
Taking this as we're going to remove this alias and use the compressed key with an ellipsis as part of 
- https://github.com/status-im/status-mobile/issues/21924

**Comment by ilmotta** [comment]
An update to my previous comment https://github.com/status-im/status-mobile/issues/21827#issuecomment-2555646750: we are discussing a few different approaches to handle the default display name which may affect the requirements from this issue.

As pointed out by @mariia-skrypnyk, what stands out to me from this issue is not so much the 3-word name being used (that's kind of harmless), but the fact that the backed up name is not being _eventually_ used.

**Comment by ulisesmac** [comment]
> but the fact that the backed up name is not being eventually used.

I think a relogin solves the issue, am I right @mariia-skrypnyk?

So this issue could focus instead in recovering the `display-name` on the first login even when the onboarding is done through keycard or recovery phrase.

**Comment by pavloburykh** [comment]
Hi @ulisesmac!

> I think a relogin solves the issue, am I right @mariia-skrypnyk?

So this issue could focus instead in recovering the display-name on the first login even when the onboarding is done through keycard or recovery phrase.

Hi @ulisesmac! Yes, re-login solves the issue in case display-name/default generated key is backed up. But in case backup has not occurred yet - 3 random name will be displayed instead of display-name/default generated key when you recover user.

Overall, the main point is that we should never display 3 random names as we have removed them some time ago from all the parts of the app.

**Comment by ulisesmac** [comment]
@pavloburykh 

Just to provide context, the display name is still generated as an `alias` on status-go everytime the user creates a new account, I'm working on fixing that 👍

**Comment by pavloburykh** [comment]
> Just to provide context, the display name is still generated 

thanks @ulisesmac. You mean `3 random names` are still generated, correct?

**Comment by ulisesmac** [comment]
@pavloburykh Yes, I meant the 3-words name is still generated as `alias` (not the display name). The `alias` is always shown when the `display-name` cannot be recovered

**Comment by pavloburykh** [comment]
> [@pavloburykh](https://github.com/pavloburykh) Yes, I meant the 3-words name is still generated as `alias` (not the display name). The `alias` is always shown when the `display-name` cannot be recovered

For such case, can we instead of 3 random name recover generated keys which a set as default until user sets up display name?

![Image](https://github.com/user-attachments/assets/7caa6c45-2787-4daa-bbf0-c5394e744443)

**Comment by ulisesmac** [comment]
We've been discussing this for some time, and we reached a solution for it. [Here's the related thread](https://discord.com/channels/1210237582470807632/1335942754152353844/1335942758308909118)
